test/common/network/io_socket_handle_impl_integration_test.cc

```diff
@@ -11,6 +11,9 @@ namespace Network {
 namespace {
 
 // Only do the integration tests in supported platforms.
+// This test requires external internet connectivity and as a result it might
+// not work under in environments that limit the external connectivity.
+// As such it is tagged with `requires-network` and is not executed in CI.
```

Comment:
cc @lizan @envoyproxy/senior-maintainers 

Tests that run on dev workstations but not in CI don't seem like good things to have.

I'm not particularly comfortable running tests locally that contact external entities.  Can we document flags that can be used to prevent tests from doing any external network communication and ideally have them be part of the default build configuration?

Reply:
> I'm not particularly comfortable running tests locally that contact external entities. Can we document flags that can be used to prevent tests from doing any external network communication and ideally have them be part of the default build configuration?

I agree, +1. I think there are a few other tests that fall into this category. For example I think there is a test for the OSX DNS resolver that requires some external connection to actually test. cc @junr03

Reply:
I agree in general, but I'd be inclined to suggest we file an issue for it and merge this clarifying comment, WDYT?

I think we can actually do non-loopback v4 with Network::Utility::getLocalAddress so I think we could have a ipv4-happy CI version but there was some docker related issue with v6 which I never tracked down (@lizan may have ideas?)

Reply:
oh actually davinci26  if you wrote this test what would you think of converting it for at least ipv4 using getLocalAddress?  bonus points if you sort out the v6 issue but I think having it on CI at all would be a big improvement.

Reply:
I'm fine with updating the comment.  I agree that we should try harder to get tcp info tests to run locally, by attempting to listen and create connections on a non-loopback interface.

Reply:
Sure makes sense, I updated the issue, https://github.com/envoyproxy/envoy/issues/15164#issuecomment-788125519 and I will follow up with tests using `getLocalAddress` instead